kiegroup/appformer
# AF-2555: Link BC <-> DB

Link BC to DB

**JIRA**: 

[AF-2555](https://issues.redhat.com/browse/AF-2555)

This is a new feature for 7.44. It links BC/Dashbuilder Webapp with Dashbuilder Runtime. It introduces in Dashbuilder Runtime the Refresh mode, which is turned on by default, and allow users from BC to call dashbuilder runtime by setting the following system properties:

* **dashbuilder.import.base.dir**. The location in Runtime configured to store dashbuilder models. In runtime the same system property is used, but it has a default value, but in BC you must set it, even if it points to the default value. Sample value: `/tmp/dashbuilder/models`;
*  **dashbuilder.runtime.location**: Dashbuilder Runtime root URL. Sample value: `http://localhost:8280`

To test it:

* Start Dashbuilder Runtime with multi mode and components enabled:

```
./bin/standalone.sh -Djboss.socket.binding.port-offset=200 -Ddashbuilder.runtime.multi=true -Ddashbuilder.components.enable=true
```
* Start BC and set the path for modes and your dashbuilder runtime location (base URL):
```
-Ddashbuilder.import.base.dir=/tmp/dashbuilder/models -Ddashbuilder.runtime.location=http://localhost:8280
```
* Then go to BC, create some content and go to Gradual Export. Close to the Download button you will see an "Open" button. Click on it and a new tab/window will open runtime with the import:


![link_with_replace](https://user-images.githubusercontent.com/359820/92292240-863f1980-eef2-11ea-91ad-c6134c0b17da.gif)
